plugins/fuse/server/server_fuse.py

```diff
@@ -422,6 +423,7 @@ def __init__(self, operations, mountpoint, *args, **kwargs):
                 'it empty?  Does the user have permission to create FUSE '
                 'mounts?  It could be another FUSE mount issue, too.' % (
                     mountpoint, ))
+            _fuseMounts.pop(name, None)
```

Comment:
This seems like an odd place to mutate global state. Is there a way we could do this in the outer thread, maybe by passing an `onError` callback?

Reply:
I suppose, ideally, this would trigger an async event so that the error handler could also join the thread.  This would involve having the FUSELogError class do an event.daemon.trigger.  The plugin would need to bind to the event (at which point it could just call `unmountGirderFuse` , possibly with an extra flag to not bother calling umount).